Tasks/Gradle/sonarqube-common/sonarqube-reportbuilder.ts

```diff
@@ -0,0 +1,100 @@
+import Q = require('q');
+
+import {SonarQubeRunSettings} from './sonarqube-runsettings';
+import {SonarQubeMetrics} from './sonarqube-metrics';
+
+import tl = require('vsts-task-lib/task');
+
+export class SonarQubeReportBuilder {
+
+    private taskMetrics:SonarQubeMetrics;
+    private sqRunSettings:SonarQubeRunSettings;
+
+    /**
+     * Creates a new SonarQubeReportBuilder, which creates Markdown-formatted reports for SonarQube analyses.
+     * @param sqRunSettings SonarQubeRunSettings object for the applicable run
+     * @param taskMetrics
+     */
+    constructor(sqRunSettings:SonarQubeRunSettings, taskMetrics:SonarQubeMetrics) {
+        if (!sqRunSettings) {
+            // Looks like: Invalid or missing task report. Check SonarQube finished successfully.
+            throw new Error(tl.loc('sqAnalysis_TaskReportInvalid'));
+        }
+
+        this.sqRunSettings = sqRunSettings;
+        this.taskMetrics = taskMetrics;
+    }
+
+
+    // Creates a string containing Markdown of a link to the SonarQube dashboard for this project.
+    /**
+     * Creates a Markdown-formatted link to the SonarQube dashboard for this project.
+     * @returns {string} A single-line, Markdown-formatted link to the SonarQube dashboard for this project
+     */
+    public createLinkToSonarQubeDashboard(): string {
```

Comment:
As stated above, the report builder should only expose a single public method. There is no need to expose the whole "sections" logic